pos_order_zip/models/order_zip_code.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class OrderZipcode(models.Model):
+    _name = "order.zipcode"
+    _rec_name = "zip_code"
+
+    zip_code = fields.Char()
```

Comment:
Why not simply build your module on top of the existing `base_location`: https://github.com/OCA/partner-contact/blob/16.0/base_location/models/res_city_zip.py

That way you ensure cross compatibility without pulling new data structures